python/pyspark/sql/context.py

```diff
@@ -433,26 +523,34 @@ def clearCache(self):
         self._ssql_ctx.clearCache()
 
     @property
-    @since(1.4)
     def read(self):
         """
         Returns a :class:`DataFrameReader` that can be used to read data
         in as a :class:`DataFrame`.
 
-        :return: :class:`DataFrameReader`
+        .. versionadded:: 1.4.0
+
+        Returns
+        -------
+        :class:`DataFrameReader`
         """
         return DataFrameReader(self)
 
     @property
-    @since(2.0)
     def readStream(self):
         """
         Returns a :class:`DataStreamReader` that can be used to read data streams
         as a streaming :class:`DataFrame`.
 
-        .. note:: Evolving.
+        .. versionadded:: 2.0.0
 
-        :return: :class:`DataStreamReader`
+        Notes
+        -----
+        This API is evolving.
```

Comment:
I switched the annotation from `.. note:: Evolving.` to this. I think user-facing doc should just better have it as prose. Looks like Python side can't use the annotation like approach unlike Scaladoc or Javadoc.

Reply:
I will document what "evolving" and "experimental" means in the doc later by referring to Scala's annotations in a separate PR.